git-hooks/pre-commit

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+import os
+import subprocess
+import sys
+from contextlib import contextmanager
+
+ABORT = 1
+
+@contextmanager
+def cd(new_dir):
+    '''Navigate to a directory in a context and then return to original directory.'''
+    prev_dir = os.getcwd()
+    os.chdir(os.path.expanduser(new_dir))
+    try:
+        yield
+    finally:
+        os.chdir(prev_dir)
+
+if __name__ == '__main__':
+    try:
+        abort = False
+        abort_messages = []
+        skip_flake8 = False
+        FLAKE8_CFG_FILE = '.flake8'
+
+        # Get base folder of repo
+        loc_proc = subprocess.Popen('git rev-parse --show-toplevel'.split(), stdout=subprocess.PIPE)
+        repo_base_folder = loc_proc.communicate()[0].decode('utf-8').strip()
+        flake8_config_path = os.path.join(repo_base_folder, FLAKE8_CFG_FILE)
+
+        # Run pytests
+#        print('_______________________' + 'Running Unit Tests' + '_______________________')
+#        with cd(repo_base_folder + '/<older>'):
+#            test_proc = subprocess.Popen(['./manage.py', 'test', '--noinput'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+#            out, err = test_proc.communicate()
+#            err_s = err.decode('utf-8')
+#            out_s = out.decode('utf-8')
+#            if 'FAILED' in err_s or 'ERROR' in err_s:
+#                abort_messages.append('All unit tests did not succeed. Please fix code/tests before committing.')
+#                abort = True
+#        print('_______________________' + 'Unit Test Output' + '_______________________')
+#        print(out_s)
+#        print('_______________________' + 'Unit Test Error' + '________________________')
+#        print(err_s)
```

Comment:
This can be used if we want unit tests to run as part of the pre-commit, thanks for pointing this out though. I'll remove for now and can add it again if we need it in the future.